src/community/elasticsearch/src/main/java/org/geoserver/elasticsearch/ElasticConfigurationPage.java

```diff
@@ -326,10 +326,19 @@ protected Component getComponentForProperty(
                                 return new Fragment(id, "empty", ElasticConfigurationPage.this);
                             }
                         } else if (property == ElasticAttributeProvider.DATE_FORMAT) {
-                            if (att.getDateFormat() != null) {
+                            List<String> validFormats = att.getValidDateFormats();
+                            if (validFormats != null) {
                                 Fragment f =
                                         new Fragment(id, "label", ElasticConfigurationPage.this);
-                                f.add(new Label("label", String.valueOf(att.getDateFormat())));
+                                String format = "";
+                                for (int i = 0; i < validFormats.size(); i++) {
+                                    if (i != valid_formats.size()) {
```

Comment:
Just noticed an error here, wrong variable name. Also working on backwards compatibility, so will have to update these things.